internal/policy/policy_test.go

```diff
@@ -68,93 +68,195 @@ func TestInitializeNamespace(t *testing.T) {
-	if err != nil {
-		t.Error(err)
-	}
+		loadedState, err = LoadState(context.Background(), repo, entry.(*rsl.ReferenceEntry))
+		if err != nil {
+			t.Error(err)
+		}
 
-	assert.Equal(t, state, loadedState)
+		assert.Equal(t, state, loadedState)
+	})
+	t.Run("fail loading while verifying multiple states, bad sig", func(t *testing.T) {
+		repo, state := createTestRepository(t, createTestStateWithPolicy)
+
+		entry, err := rsl.GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		loadedState, err := LoadState(context.Background(), repo, entry.(*rsl.ReferenceEntry))
+		if err != nil {
+			t.Error(err)
+		}
+
+		assert.Equal(t, state, loadedState)
+
+		targetsMetadata, err := state.GetTargetsMetadata(TargetsRoleName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		targetsMetadata, err = AddDelegation(targetsMetadata, "test-rule-1", []*tuf.Key{}, []string{""}, 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+		state.ruleNames.Add("test-rule-1")
+
+		env, err := dsse.CreateEnvelope(targetsMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		signer, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(rootKeyBytes) //nolint:staticcheck
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		badSigner, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(targets1KeyBytes) //nolint:staticcheck
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		env, err = dsse.SignEnvelope(context.Background(), env, signer)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		state.TargetsEnvelope = env
+
+		if err := state.Commit(repo, "", false); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := Apply(context.Background(), repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		targetsMetadata, err = AddDelegation(targetsMetadata, "test-rule-2", []*tuf.Key{}, []string{""}, 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+		state.ruleNames.Add("test-rule-2")
+
+		env, err = dsse.CreateEnvelope(targetsMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		env, err = dsse.SignEnvelope(context.Background(), env, badSigner)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		state.TargetsEnvelope = env
+
+		if err := state.Commit(repo, "", false); err != nil {
+			t.Fatal(err)
+		}
+		policyStagingRef, err := repo.Reference(plumbing.ReferenceName(PolicyStagingRef), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		newPolicyRef := plumbing.NewHashReference(PolicyRef, policyStagingRef.Hash())
+		if err := repo.Storer.SetReference(newPolicyRef); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := rsl.NewReferenceEntry(PolicyRef, policyStagingRef.Hash()).Commit(repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		entry, err = rsl.GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		loadedState, err = LoadState(context.Background(), repo, entry.(*rsl.ReferenceEntry))
+
+		assert.Equal(t, "requested state has invalidly signed metadata: verifier's key and threshold constraints not met", err.Error())
```

Comment:
I know the text is from the `fmt.Errorf` but I think the embedded error is named. Can we just do assert.ErrorIs on that one? Probably some thing about verifier constraints not being met or so

Reply:
how would I be able to compare that single part of the error?

Reply:
Does assert.ErrorIs not match nested errors?